pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -70,34 +76,34 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return reconcile.Result{}, err
 	}
 
-	if !instance.Spec.Features.ReconcilePullSecret {
-		return reconcile.Result{}, nil
-	}
-
-	operatorSecret, err := r.kubernetescli.CoreV1().Secrets(operator.Namespace).Get(ctx, operator.SecretName, metav1.GetOptions{})
-	if err != nil {
+	userSecret, err := r.kubernetescli.CoreV1().Secrets(pullSecretName.Namespace).Get(ctx, pullSecretName.Name, metav1.GetOptions{})
+	if err != nil && !kerrors.IsNotFound(err) {
 		return reconcile.Result{}, err
 	}
 
-	var userSecret *corev1.Secret
+	if !instance.Spec.OperatorFlags.GetSimpleBoolean(ENABLED) {
```

Comment:
Should this check for enabled happen before any check for secrets?

Reply:
done